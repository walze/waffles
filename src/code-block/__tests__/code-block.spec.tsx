import { render, fireEvent } from '@testing-library/react';

import { CodeBlock } from '../index';

const sizes = ['small', 'medium', 'large'] as const;

describe('CodeBlock', () => {
  it('renders pre and code elements containing the text', () => {
    const { container } = render(
      <CodeBlock>{`pyftsubset JetBrainsMono.ttf
 --output-file="JetBrainsMono.woff"
 --flavor=woff
 --with-zopfli
 --layout-features="kern,liga,clig,ccmp"
 --unicodes="*"`}</CodeBlock>,
    );

    const preElement = container.querySelector('pre');
    const codeElement = container.querySelector('code');

    expect(preElement).toBeInTheDocument();
    expect(preElement).toContainElement(codeElement);
    expect(codeElement).toBeInTheDocument();
    expect(codeElement).toHaveTextContent(/JetBrainsMono.woff/i);
  });

  it('sets the data attribute on the pre element', () => {
    const { getByTestId } = render(
      <CodeBlock data-testid="code-block-element">
        npm install @datacamp/waffles
      </CodeBlock>,
    );

    const preElement = getByTestId('code-block-element');

    expect(preElement).toBeInTheDocument();
  });

  it('renders snapshot of focused state', () => {
    const { getByTestId } = render(
      <CodeBlock data-testid="code-block-element">test</CodeBlock>,
    );

    const preElement = getByTestId('code-block-element');
    fireEvent.focus(preElement);

    expect(preElement).toMatchSnapshot();
  });

  describe('renders snapshot of', () => {
    sizes.forEach((size) => {
      it(`size ${size}`, () => {
        const { container } = render(<CodeBlock size={size}>test</CodeBlock>);

        const codeElement = container.querySelector('pre');
        expect(codeElement).toMatchSnapshot();
      });
    });
  });

  describe('renders snapshot of inverted', () => {
    sizes.forEach((size) => {
      it(`size ${size}`, () => {
        const { container } = render(
          <CodeBlock size={size} inverted>
            test
          </CodeBlock>,
        );

        const codeElement = container.querySelector('pre');
        expect(codeElement).toMatchSnapshot();
      });
    });
  });
});
