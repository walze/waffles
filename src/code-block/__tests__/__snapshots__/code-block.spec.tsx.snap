// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock renders snapshot of focused state 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: multiply;
  font-size: 12px;
  line-height: 1.5;
  color: #05192D;
  background-color: #EFEBE4;
  padding: 8px;
  margin: 0;
  tab-size: 4;
  white-space: pre;
  overflow-x: auto;
  outline: 0;
  box-shadow: 0 0 0 2px #009BD8;
}

<pre
  class="emotion-0"
  data-testid="code-block-element"
  tabindex="0"
>
  <code>
    test
  </code>
</pre>
`;

exports[`CodeBlock renders snapshot of inverted size large 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: regular;
  font-size: 14px;
  line-height: 1.5;
  color: #FFFFFF;
  background-color: rgba(255, 255, 255, 0.3);
  padding: 8px;
  margin: 0;
  tab-size: 4;
  white-space: pre;
  overflow-x: auto;
  outline: 0;
}

<pre
  class="emotion-0"
  tabindex="0"
>
  <code>
    test
  </code>
</pre>
`;

exports[`CodeBlock renders snapshot of inverted size medium 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: regular;
  font-size: 12px;
  line-height: 1.5;
  color: #FFFFFF;
  background-color: rgba(255, 255, 255, 0.3);
  padding: 8px;
  margin: 0;
  tab-size: 4;
  white-space: pre;
  overflow-x: auto;
  outline: 0;
}

<pre
  class="emotion-0"
  tabindex="0"
>
  <code>
    test
  </code>
</pre>
`;

exports[`CodeBlock renders snapshot of inverted size small 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: regular;
  font-size: 10px;
  line-height: 1.5;
  color: #FFFFFF;
  background-color: rgba(255, 255, 255, 0.3);
  padding: 8px;
  margin: 0;
  tab-size: 4;
  white-space: pre;
  overflow-x: auto;
  outline: 0;
}

<pre
  class="emotion-0"
  tabindex="0"
>
  <code>
    test
  </code>
</pre>
`;

exports[`CodeBlock renders snapshot of size large 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: multiply;
  font-size: 14px;
  line-height: 1.5;
  color: #05192D;
  background-color: #EFEBE4;
  padding: 8px;
  margin: 0;
  tab-size: 4;
  white-space: pre;
  overflow-x: auto;
  outline: 0;
}

<pre
  class="emotion-0"
  tabindex="0"
>
  <code>
    test
  </code>
</pre>
`;

exports[`CodeBlock renders snapshot of size medium 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: multiply;
  font-size: 12px;
  line-height: 1.5;
  color: #05192D;
  background-color: #EFEBE4;
  padding: 8px;
  margin: 0;
  tab-size: 4;
  white-space: pre;
  overflow-x: auto;
  outline: 0;
}

<pre
  class="emotion-0"
  tabindex="0"
>
  <code>
    test
  </code>
</pre>
`;

exports[`CodeBlock renders snapshot of size small 1`] = `
.emotion-0 {
  font-family: JetBrainsMonoNL,Menlo,Monaco,'Courier New',monospace;
  font-weight: 400;
  border-radius: 4px;
  margin: 0 2px;
  padding: 2px 4px;
  white-space: nowrap;
  mix-blend-mode: multiply;
  font-size: 10px;
  line-height: 1.5;
  color: #05192D;
  background-color: #EFEBE4;
  padding: 8px;
  margin: 0;
  tab-size: 4;
  white-space: pre;
  overflow-x: auto;
  outline: 0;
}

<pre
  class="emotion-0"
  tabindex="0"
>
  <code>
    test
  </code>
</pre>
`;
